emsdk.py

```diff
@@ -1885,6 +1885,15 @@ def install_tool(self):
       print("Skipped installing " + self.name + ", already installed.")
       return True
 
+    version_hash = self.name
+    try:
```

Comment:
Can you just use `if os.path.exists()` here instead of try catch?

Reply:
Isn't that racey?  E.g. someone decides to try to clean up their folder while running this command and it then fails because the file existed when we checked but now it doesn't.

Reply:
It is a minor TOCTOU violation yes.  However, I don't mind crashing on people try to install/upgrade and SDK while that the same time running `rm -rf` on it.   I think its OK, maybe even better, to crash in such cases. 

We generally try to avoid try/catch block for this kind of thing in emscripten.   There are many of these kinds of TOCTOU (many that we can't protect against) already in the codebase.  e.g. checking the llvm version before calling in.

Reply:
Updated to use `isfile`.  Also did some minor cleanup of variable usage.